### PR TITLE
Service Manager V2/V3 compatibility

### PR DESCRIPTION
**Pull request by marc-mabe**
This component is a `require-dev` for zend-cache - see zendframework/zend-cache#64

### CODE REVIEW

File: src/AdapterPluginManager.php [hunk 1]
```diff
@@ -39,14 +40,53 @@ class AdapterPluginManager extends AbstractPluginManager
...
+        Adapter\MsgPack::class              => InvokableFactory::class,
+        'zendserializeradaptermsgpack'      => InvokableFactory::class,
+        Adapter\PhpCode::class              => InvokableFactory::class,
+        'zendserializeradapter\phpcode'     => InvokableFactory::class,
+        Adapter\PhpSerialize::class         => InvokableFactory::class,
+        'zendserializeradapterphpserialize' => InvokableFactory::class,
+        Adapter\PythonPickle::class         => InvokableFactory::class,
+        'zendserializeradapterpythonpickle' => InvokableFactory::class,
+        Adapter\Wddx::class                 => InvokableFactory::class,
+        'zendserializeradapterwddx'         => InvokableFactory::class
```
Comment (marc-mabe) [inline src/AdapterPluginManager.php:56]: @weierophinney @ezimuel Is that really the way to go ?

**Comment by ezimuel** [comment]
@marc-mabe thanks!

File: src/AdapterPluginManager.php (hunk 1 above)
Comment (weierophinney) [inline src/AdapterPluginManager.php:56]: No. You should have a combination of aliases and factories. InvokableFactory will not work with the above aliases; it assumes the requested name is the class name. What you should do is:
- map each FCQN for a plugin to an `InvokableFactory`
- map aliases for each plugin to the FQCN. Typically, you'll have multiple variants, but typically all will be in _short form_. You don't have to do normalized versions of the FQCN as you did above.

Comment (weierophinney) [inline src/AdapterPluginManager.php:56]: I've opened #11 to demonstrate. It looks like @ezimuel added a number of aliases when he merged your changes; I've added a few more. I find that we typically need each of:
- lowercase
- camelCase
- TitleCase

to hit all the various permutations used be folks when fetching from the plugin managers.

**Pull request merged**